components/brave_ads/core/internal/settings/settings.cc

```diff
@@ -31,6 +31,11 @@ bool UserHasOptedInToNewTabPageAds() {
                  kNewTabPageShowSponsoredImagesBackgroundImage);
 }
 
+bool UserHasOptedInToSearchAds() {
+  return UserHasJoinedBraveRewards() &&
```

Comment:
This should be removed from here because the ads library should run business logic for both non-rewards and rewards users. `UserHasJoinedBraveRewards()` should be moved to where we decide the default value for `prefs::kOptedInToSearchAds`